components/layout_2020/flexbox/layout.rs

```diff
@@ -844,9 +867,11 @@ impl FlexLine<'_> {
         &mut self,
         flex_context: &mut FlexContext,
         container_main_size: Au,
+        main_gap: Au,
     ) -> FlexLineLayoutResult {
+        let item_count = self.items.len();
         let (item_used_main_sizes, mut free_space) =
-            self.resolve_flexible_lengths(container_main_size);
+            self.resolve_flexible_lengths(container_main_size - main_gap * (item_count as i32 - 1));
```

Comment:
...but here it might be possible to have zero items, then adding the gap seems wrong, but not sure if it matters.